src/components/services/Transmission.vue

```diff
@@ -0,0 +1,194 @@
+        >{{ count }}
+        <template v-if="count === 1">torrent</template>
+        <template v-else>torrents</template>
+      </span>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import service from "@/mixins/service.js";
+import Generic from "./Generic.vue";
+
+const units = ["B", "KB", "MB", "GB"];
+
+// Take the rate in bytes and keep dividing it by 1k until the lowest
+// value for which we have a unit is determined. Return the value with
+// up to two decimals as a string and unit/s appended.
+const displayRate = (rate) => {
+  let i = 0;
+
+  while (rate > 1000 && i < units.length) {
+    rate /= 1000;
+    i++;
+  }
+  return (
+    Intl.NumberFormat(undefined, { maximumFractionDigits: 2 }).format(
+      rate || 0
+    ) + ` ${units[i]}/s`
+  );
+};
+
+export default {
+  name: "Transmission",
+  mixins: [service],
+  props: { item: Object },
+  components: { Generic },
+  data: () => ({ dl: null, ul: null, count: null, error: null }),
+  computed: {
+    downRate: function () {
+      return displayRate(this.dl);
+    },
+    upRate: function () {
+      return displayRate(this.ul);
+    },
+  },
+  created() {
+    const updateInterval = parseInt(this.item?.updateInterval, 10) || 0;
+    if (updateInterval > 0) {
+      this.interval = setInterval(() => this.fetchData(), updateInterval);
+    }
+    if (this.item?.sessionId) {
+      this.sessionId = this.item.sessionId;
+    }
+
+    this.fetchData();
+  },
+  unmounted() {
+    if (this.interval) {
+      clearInterval(this.interval);
+    }
+  },
+  methods: {
+    fetchData: async function () {
+      try {
+        const auth = {};
+        if (this.item?.username && this.item?.password) {
+          auth.user = this.item.username;
+          auth.password = this.item.password;
+        }
+        const {
+          arguments: { activeTorrentCount, downloadSpeed, uploadSpeed },
+        } = await this.transmissionFetch("session-stats", auth);
+
+        this.error = false;
+        this.ul = uploadSpeed;
+        this.dl = downloadSpeed;
+        this.count = activeTorrentCount;
+      } catch (e) {
+        this.error = true;
+        console.error(e);
+      }
+    },
+    /**
+     * @param {string} method
+     * @param {{user: string, password: string}?} auth
+     * @returns {Promise<{result: string, arguments: any}>}
+     */
+    transmissionFetch: async function (method, auth) {
+      /**
+       * @type {RequestInit}
+       */
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          method: method,
+        }),
+      };
+
+      if (this.proxy?.useCredentials) {
+        options.credentials = "include";
+      }
+
+      // Each item can override the credential settings
+      if (this.item.useCredentials !== undefined) {
+        options.credentials =
+          this.item.useCredentials === true ? "include" : "omit";
+      }
```

Comment:
A [custom fetch](https://github.com/bastienwirtz/homer/blob/main/src/mixins/service.js) function is available (you already included it using the `service` mixin), which handle the credential option and more. Could you update your code to use it and avoid code duplication ? Thanks :pray:

Reply:
@bastienwirtz 

With the existing custom fetch, we don't have access to the response in the event of a non Response.OK result.  We would need access to the response in order to obtain the  "X-Transmission-Session-Id" after we are sent a 409 status code.

I have hacked it trivially with, but I'm sure there is a better / more correct way.

In service.js
```
`fetch: function (path, init, json = true, forceResponse = false)`
...
...
        if (!response.ok) {
          if (!forceResponse)
            throw new Error("Not 2xx response");
          else
            return response;
        }
```